### PR TITLE
CHE-10883 migrate from bower to yarn

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Migrate from bower to yarn.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10881
https://github.com/eclipse/che/pull/10883

### CODE REVIEW

File: dashboard/Dockerfile [hunk 1]
```diff
@@ -19,10 +19,9 @@ RUN apt-get update && \
     && apt-get -y clean \
     && rm -rf /var/lib/apt/lists/*
 COPY package.json /dashboard/
-RUN cd /dashboard && npm install
-COPY bower.json /dashboard/
-RUN cd /dashboard && ./node_modules/.bin/bower install --allow-root
+RUN cd /dashboard && npm i yarn && node_modules/yarn/bin/yarn install --ignore-scripts
```
Comment (benoitf) [inline dashboard/Dockerfile]: why do we need to run yarn a first time ?

Comment (akurinnoy) [reply dashboard/Dockerfile]: I suppose to install dependencies, and the second run is to run build

Comment (olexii4) [reply dashboard/Dockerfile]: Sorry. I'm just having a look at it.
Yes. @akurinnoy is a quite right

**Comment by ashumilova** [comment]
ci-test

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/842/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/842//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/847/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/847//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/850/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/850//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/852/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/852//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/853/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/853//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by musienko-maxim — APPROVED** [review]

**Comment by musienko-maxim** [comment]
The functional parts work as well. But we have broken icons and dependent images on dashboard. See screenshots below:
workspacedetailsoverviewtest shouldchecknamefield_time-1537882597719-millis

**Review by musienko-maxim — CHANGES_REQUESTED** [review]
We should fix images and icons

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/859/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/859//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/871/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/871//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by olexii4** [comment]
ci-test

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/888/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/888//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by olexii4** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/894/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/894//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by SkorikSergey** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/894/) doesn't show any regression against this Pull Request.

**Review by musienko-maxim — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

File: dashboard/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+export interface ICheWindow extends Window{
```
Comment (benoitf) [inline dashboard/src/index.ts]: missing space after Window

**Review by akurinnoy — APPROVED** [review]

**Comment by nickboldt** [comment]
Will this change be included in 6.12.0 release?

**Comment by nickboldt** [comment]
Also I just noticed this moves us from node 6.11.2 -> 8.10.0.

Will a similar change also happen for the node project in https://github.com/eclipse/che-lib/tree/master/che-terminal-client ? It would be good to be able to build all of Che with the same pod definition (all with the same version of npm/node/mvn/java/golang).

**Comment by olexii4** [comment]
@AndrienkoAleksandr  WDYT?

**Comment by AndrienkoAleksandr** [comment]
@nickboldt for build xterm.js we don't use docker.

**Comment by nickboldt** [comment]
> @nickboldt for build xterm.js we don't use docker.

I'm not talking about docker - I asked about node 6 -> 8. You definitely use npm:

https://github.com/eclipse/che-lib/blob/master/che-terminal-client/pom.xml#L88

**Comment by AndrienkoAleksandr** [comment]
@nickboldt I do full rebuild Eclipse CHE quite often. And On the my laptop I set up node 8.9.1. So I do build  xterm.js (in the che-lib)  with this version and everithing is ok.

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/918/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/918//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11301
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by SkorikSergey — APPROVED** [review]

**Pull request merged**